packages/ember-routing-views/lib/views/link.js

```diff
@@ -320,7 +320,7 @@ var LinkComponent = EmberComponent.extend({
     }
 
     var routing = get(this, '_routing');
-    var targetRouteName = get(this, 'targetRouteName');
+    var targetRouteName = this._handleOnlyQueryParamsSupplied(get(this, 'targetRouteName'));
```

Comment:
Yes it can be renamed to something more meaningful like `computeWithOnlyQueryParamsSupplied`.